ext/http/00_serve.js

```diff
@@ -534,6 +534,7 @@ function mapToCallback(responseBodies, context, signal, callback, onError) {
 
     // Did everything shut down while we were waiting?
     if (context.closed) {
+      op_http_set_promise_complete(req, 500);
```

Comment:
Why 500 here?

Reply:
We're shutting down, so this status shouldn't make it back to the client. Will change to 503 and add a comment.